specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-04-01-preview/ContentPackages.json

```diff
@@ -0,0 +1,498 @@
+      "get": {
+        "x-ms-examples": {
+          "Get all available packages.": {
+            "$ref": "./examples/contentPackages/GetPackageById.json"
+          }
+        },
+        "tags": [
+          "ContentPackages"
+        ],
+        "description": "Gets an installed packages by its id.",
+        "operationId": "ContentPackages_Get",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/ODataFilter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/ODataOrderBy"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/ODataTop"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/ODataSkipToken"
+          },
+          {
+            "$ref": "#/parameters/PackageIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/packageModel"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "x-ms-examples": {
+          "Install a package to the workspace.": {
+            "$ref": "./examples/contentPackages/InstallPackage.json"
+          }
+        },
+        "tags": [
+          "ContentPackages"
+        ],
+        "description": "Install a package to the workspace.",
+        "operationId": "ContentPackage_Install",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/PackageIdParameter"
+          },
+          {
+            "$ref": "#/parameters/PackageInstallationProperties"
```

Comment:
>PackageInstallationProperties [](http://example.com/codeflow?start=34&length=29)

the incoming PUT needs to have the full resource not just the properties in the body

Reply:
Ideally this should be packagemodel

Reply:
@raosuhas  It was package model. But I got an error from the scanner says the package Id should be removed because it is already in the url. So I removed it to satisfy the scanner.

Reply:
Since packageId is part of "id" property of packageModel anyway, why is packageId property needed explicitly needed in packageModel? That would eliminate the need of separate model for request and response.

Also, ARM requires RPs to have same model for request and response in PUT APIs.

Reply:
@visingla-ms 
One of the scanner, I think it is the ~[Staging] Swagger LintDiff scanner, suggests me to remove PackageIdParameter in the body as it is already in the URL. If I should ignore it. please let me know.

Reply:
Sorry, my question was, why not remove "packageId" property from "packageModel" itself. Then if you will use 
packageModel" in request payload also, Linter won't have any issue. And, you don't "packageId" in "packageModel", as it is already part of "id" property of "packageModel"

Reply:
@visingla-ms 
I believe there is another rule. If the get by id API return a response object, the list APIs should return a collection of the same object.  The packageId is our identifier of response object.  A user would need this id to call get by id from the list in our workflow.

Reply:
@xuhumsft Let's catch up offline for this point. I am finding more details about usage of free-form objects, will ping you after that.

Reply:
Hi @visingla-ms,
I have a discussion with the same.  We would change the packageId to contentId and we still need the conentId in the body. 
The contentId can be the same as the {id} part of the url but It is also possible that they are not equal.
In some cases, we allow the contentId to be the same with different contentKind. In that case, the {id} in the url would have those information to make sure no collisions. However, we still need the conentId for searching. So, we decide to have the decidated contentId in the body.  the {id} could be different to avoid collisions.